src/core/dyad_open_dlwrap.c

```diff
@@ -0,0 +1,135 @@
+#include "dyad_open_dlwrap.h"
```

Comment:
The file name can change to io_interceptor or something else because  it is not only for open but also for close

Reply:
Actually, I've phased this out in favor of `reenter`. I'm keeping the files around until I know everything is working, just to be safe. Assuming things work as expected, these files will be deleted.